app/scripts/modules/core/loadBalancer/loadBalancer.write.service.js

```diff
@@ -32,7 +32,7 @@ module.exports = angular
     function upsertLoadBalancer(loadBalancer, application, descriptor, params = {}) {
       var name = loadBalancer.clusterName || loadBalancer.name;
       loadBalancer.cloudProvider = loadBalancer.provider;
-      if (loadBalancer.healthCheckProtocol.indexOf('HTTP') === 0) {
+      if (loadBalancer.healthCheckProtocol != null && loadBalancer.healthCheckProtocol.indexOf('HTTP') === 0) {
```

Comment:
If you want to catch all "falsy" values (`false`, `undefined`, `null`, `''`, `0`), you can omit the infix operator check altogether, i.e. `if (loadBalancer.healthCheckProtocol)` is sufficient. In JS, you generally want to avoid `!=` and `==`, since the JS engine will coerce one of the arguments to be the same type as the other argument. You can get into a lot of confusing situations, so it's considered best practice to perform a strict equality check via `!==` or `===`, or perform a boolean coercion by omitting the implicit check (e.g. `if (loadBalancer.healthCheckProtocol) {`.

This might read a little cleaner for this code block, assuming `healthCheckProtocol` is an optional field but will never be an empty string:
```javascript
let protocol = loadBalancer.healthCheckProtocol || ''; // if the field is optional, the || can be omitted
if (protocol.startsWith('HTTP')) {
  loadBalancer.healthCheck = `${protocol}:${loadBalancer.healthCheckPort}${loadBalancer.healthCheckPath}`;
} else {
  loadBalancer.healthCheck = `${protocol}:${loadBalancer.healthCheckPort}`;
}
```